src/coreclr/src/tools/aot/ILCompiler.TypeSystem.ReadyToRun.Tests/SignatureTests.cs

```diff
@@ -27,6 +27,42 @@ public SignatureTests()
             _testModule = _context.GetModuleForSimpleName("ILTestAssembly");
         }
 
+        private static string GetModOptMethodSignatureInfo(MethodSignature signature)
+        {
+            if (!signature.HasEmbeddedSignatureData || signature.GetEmbeddedSignatureData() == null)
+                return "";
+
+            StringBuilder sb = new StringBuilder();
+            foreach (EmbeddedSignatureData data in signature.GetEmbeddedSignatureData())
+            {
+                sb.Append(data.kind.ToString());
+                sb.Append(data.index);
+                sb.Append(data.type.ToString());
```

Comment:
`TypeDesc.ToString` is not considered stable. I would rather not introduce dependencies on it.